### DISCUSSION TITLE
New to RLS, trouble with setting policies

### DISCUSSION THREAD

**Discussion by chen-rn** [supabase/supabase#8352] [Questions] [answered]
context: i'm using prisma + supabase

**what i'm trying to do**
I'm trying to add a the following policy to my Image table.
`auth.uid() = userId`
I would like all images to have a userId to show who the owner is.

**This is the error I'm getting from Supabase**
<img width="500" alt="Screen Shot 2022-08-15 at 2 46 10 PM" src="https://user-images.githubusercontent.com/36214945/184723955-ccc0ef15-2fed-4240-90a5-41f646e59fda.png">

But i do have a user id field as you can see in my Supabase
<img width="500" alt="image" src="https://user-images.githubusercontent.com/36214945/184723918-875d08ec-5ca7-4347-bdc5-29673b0ff95c.png">
(does the fact that it's a relational field matter?)

Here's my prisma.schema file. Not sure if I'm doing it wrong.
```
model User {
    id        String   @id @default(uuid()) @db.Uuid
    email     String   @unique
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt
    Image     Image[]
}

model Image {
    id        String   @unique @default(uuid())
    url       String
    createdAt DateTime @default(now())
    updatedAt DateTime @updatedAt
    user      User     @relation(fields: [userId], references: [id])
    userId    String   @db.Uuid
}
```

I've followed [this](https://www.youtube.com/watch?v=Ow_Uzedfohk) Youtube tutorial on policies but to be frank I'm still confused as to what are the variables that you can access while writing the inputs.

**Comment by GaryAustin1** [comment, marked as answer]
You are using capital letters which is a bad idea in Postgres, unless you are forced to and using Prisma seems to be one of those cases.

When you use capital letters you MUST enclose all names in double quotes in SQL.  So you need auth.uid() = "userId".  

Any table names you use will also need to be in double quotes.   

This is not to be confused with text string in Postgres which are in single quotes.

**Reply by chen-rn** [reply]
WOW! IT WORKED!!!! I would've never guessed that!

Thank you so much! 🎉🎉🎉🎉🎉🎉
